ngx_http_auth_ldap_module.c

```diff
@@ -431,18 +436,24 @@ ngx_http_auth_ldap_ldap_server(ngx_conf_t *cf, ngx_command_t *dummy, void *conf)
             return NGX_CONF_ERROR;
         }
         server->connections = i;
-    } else if (ngx_strcmp(value[0].data, "ssl_check_cert") == 0  && ngx_strcmp(value[1].data, "on") == 0) {
-      #if OPENSSL_VERSION_NUMBER >= 0x10002000
-      server->ssl_check_cert = 1;
-      #else
-      #if GNUC > 4
-      #warning "http_auth_ldap: Compiling with OpenSSL < 1.0.2, certificate verification will be unavailable. OPENSSL_VERSION_NUMBER == " XSTR(OPENSSL_VERSION_NUMBER)
-      #endif
-      ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
-        "http_auth_ldap: 'ssl_cert_check': cannot verify remote certificate's domain name because "
-        "your version of OpenSSL is too old. "
-        "Please install OpenSSL >= 1.02 and recompile nginx.");
-      #endif
+    } else if (ngx_strcmp(value[0].data, "ssl_check_cert") == 0) {
+        #if OPENSSL_VERSION_NUMBER >= 0x10002000
+        if ((ngx_strcmp(value[1].data, "on") == 0) || (ngx_strcmp(value[1].data, "full") == 0)) {
+            server->ssl_check_cert = SSL_CERT_VERIFY_FULL;
+        } else if (ngx_strcmp(value[1].data, "chain") == 0) {
+            server->ssl_check_cert = SSL_CERT_VERIFY_CHAIN;
+        } else {
+            server->ssl_check_cert = SSL_CERT_VERIFY_OFF;
+        }
+        #else
+        #if GNUC > 4
+        #warning "http_auth_ldap: Compiling with OpenSSL < 1.0.2, certificate verification will be unavailable. OPENSSL_VERSION_NUMBER == " XSTR(OPENSSL_VERSION_NUMBER)
+        #endif
+        ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+          "http_auth_ldap: 'ssl_cert_check': cannot verify remote certificate's domain name because "
+          "your version of OpenSSL is too old. "
+          "Please install OpenSSL >= 1.0.2 and recompile nginx.");
+        #endif
```

Comment:
Leave directives for checking `OPENSSL_VERSION_NUMBER` in place. If `ssl_check_cert` is specified, `on` and `full` indicate full certificate checking (chain and host/IP), `chain` indicates checking certificate chain only but not host/IP, and anything else means `off` (don't verify the certificate).